lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -242,7 +242,7 @@ public function fetch($fetchMode = null)
     {
         $values = $this->_fetch();
         if (null === $values) {
-            return null;
```

Comment:
I'm not sure that I understand your question.
It's supposed to return false or an array (boolean|array).

Now that I think about it, _fetch is not even supposed to return null in the first place, so that should be fixed as well I suppose.